libc/string/string_cmp.c

```diff
@@ -0,0 +1,439 @@
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp("\0", charEdgeValue, 1));
+}
+
+
+TEST(string_memcmp, emptyInput)
+{
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp(empty, asciiSet, BUFF_SIZE));
+	TEST_ASSERT_GREATER_THAN_INT(0, memcmp(asciiSet, empty, BUFF_SIZE));
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp(empty, nonZeroAsciiSet, BUFF_SIZE));
+	TEST_ASSERT_GREATER_THAN_INT(0, memcmp(nonZeroAsciiSet, empty, BUFF_SIZE));
+	TEST_ASSERT_EQUAL_INT(0, memcmp(empty, empty, BUFF_SIZE));
+
+	/* Memory cmp is not sensitive for NUL characters */
+	TEST_ASSERT_NOT_EQUAL_INT(0, memcmp(empty, separated, sizeof(separated)));
+}
+
+
+TEST(string_memcmp, big)
+{
+	TEST_ASSERT_EQUAL_INT(0, memcmp(hugeString, hugeString, BUFF_SIZE * 4));
+	TEST_ASSERT_LESS_THAN_INT(0, memcmp(hugeZeroString, hugeString, BUFF_SIZE * 4));
+	TEST_ASSERT_GREATER_THAN_INT(0, memcmp(hugeString, hugeZeroString, BUFF_SIZE * 4));
+	TEST_ASSERT_EQUAL_INT(0, memcmp(hugeZeroString, hugeZeroString, BUFF_SIZE * 4));
+}
+
+
+TEST(string_memcmp, various_sizes)
+{
+	int i;
+
+	for (i = 1; i < BUFF_SIZE; i++) {
+		/* Checking various sizes of compare passed to memcmp */
+		TEST_ASSERT_LESS_THAN_INT(0, memcmp(asciiSet, nonZeroAsciiSet, i));
+
+		/*
+		 * The testing compares elements that value are every time on a different spot
+		 * and will check if the placement of the dataset on s1 and s2 will not impact on this
+		 */
+		support[i] = 1;
+		TEST_ASSERT_LESS_THAN_INT(0, memcmp(empty, support, BUFF_SIZE));
+		TEST_ASSERT_GREATER_THAN_INT(0, memcmp(support, empty, BUFF_SIZE));
+		support[i] = 0;
+	}
+}
+
+
+TEST(string_memcmp, offsets)
+{
+
+	char dataSet[4000] = { 0 },
+		 supportSet[4000] = { 0 };
+	int s1Offs, s2Offs, szOffset, sz = sizeof(dataSet);
+
+	for (s1Offs = 0; s1Offs < sz; s1Offs++) {
+		dataSet[s1Offs] = s1Offs;
+		supportSet[s1Offs] = s1Offs;
+	}
+
+	/*Testing different offset of data blocks with same space or different*/
+	for (s1Offs = 0; s1Offs < 8; s1Offs++) {
+		for (s2Offs = 0; s2Offs < 8; s2Offs++) {
+			for (szOffset = 0; szOffset < 8; szOffset++) {
+
+				if (s2Offs < s1Offs) {
+					TEST_ASSERT_GREATER_THAN_INT(0, memcmp(&dataSet[s1Offs], &dataSet[s2Offs], sz - (s1Offs + szOffset)));
+					TEST_ASSERT_GREATER_THAN_INT(0, memcmp(&dataSet[s1Offs], &supportSet[s2Offs], sz - (s1Offs + szOffset)));
+				}
+				else if (s2Offs == s1Offs) {
+					TEST_ASSERT_EQUAL_INT(0, memcmp(&dataSet[s1Offs], &dataSet[s2Offs], sz - (s1Offs + s2Offs + szOffset)));
+					TEST_ASSERT_EQUAL_INT(0, memcmp(&dataSet[s1Offs], &supportSet[s2Offs], sz - (s1Offs + s2Offs + szOffset)));
+				}
+				else {
+					TEST_ASSERT_LESS_THAN_INT(0, memcmp(&dataSet[s1Offs], &dataSet[s2Offs], sz - (s2Offs + szOffset)));
+					TEST_ASSERT_LESS_THAN_INT(0, memcmp(&dataSet[s1Offs], &supportSet[s2Offs], sz - (s2Offs + szOffset)));
+				}
+			}
+		}
+	}
+}
+
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_strncmp)
```

Comment:
Please apply the suggestions above to the rest of functions too, in case of doubts please let me know.